services/frontend/src/components/admin/skillTable/SkillTable.jsx

```diff
@@ -24,52 +24,35 @@ const SkillTable = ({ intl }) => {
   const [selectedRowKeys, setSelectedRowKeys] = useState([]);
   const axios = useAxios();
   const history = useHistory();
-
   const dispatch = useDispatch();
 
-  // Fetches the skill information
-  const getSkill = useCallback(async () => {
+  const getBackendInfo = useCallback(async () => {
     try {
       dispatch(setAdminSkillsLoading(true));
-
-      const results = await axios.get(`api/option/skillsAllLang`);
-
-      // Formats data from backend into viewable data for the table
-      const formattedData = results.data.map((category) => ({
-        ...category,
-        key: category.id,
-      }));
-
-      dispatch(setAdminSkills(formattedData));
-    } catch (error) {
-      handleError(error, "redirect", history);
-    }
-  }, [axios, dispatch, history]);
-
-  // Fetches the category information
-  const getCategories = useCallback(async () => {
-    try {
       dispatch(setAdminCategoriesLoading(true));
-
-      const results = await axios.get(`api/option/categoriesAllLang`);
-
-      // Formats data from backend into viewable data for the table
-      const formattedData = results.data.map((category) => ({
-        ...category,
-        key: category.id,
-      }));
-
-      dispatch(setAdminCategories(formattedData));
+      Promise.all([
+        axios.get(`api/option/skillsAllLang`),
+        axios.get(`api/option/categoriesAllLang`),
+      ]).then(([skill, categories]) => {
+        dispatch(
+          setAdminSkills(
+            skill.data.map((category) => ({ ...category, key: category.id }))
+          )
+        );
+        dispatch(
+          setAdminCategories(
+            categories.data.map((category) => ({
+              ...category,
+              key: category.id,
+            }))
+          )
+        );
+      });
```

Comment:
since we are using catch the async way, not the promise way here, could we maybe use awaits so its less nested?

```suggestion
      const [skill, categories] = await Promise.all([
        axios.get(`api/option/skillsAllLang`),
        axios.get(`api/option/categoriesAllLang`),
      ]);
      
      dispatch(
        setAdminSkills(
          skill.data.map((category) => ({ ...category, key: category.id }))
        )
      );
      
      dispatch(
        setAdminCategories(
          categories.data.map((category) => ({
            ...category,
            key: category.id,
          }))
        )
      );
```